src/workerd/api/web-socket.c++

```diff
@@ -583,9 +583,14 @@ void WebSocket::ensurePumping(jsg::Lock& js) {
         // this didn't cause our writes to fail, maybe due to timing. Let's set the error now.
         reportError(js, KJ_EXCEPTION(DISCONNECTED, "WebSocket peer disconnected"));
       } else if (native.closedIncoming && native.closedOutgoing) {
-        // Native WebSocket no longer needed; release.
-        KJ_ASSERT(native.state.is<Accepted>());
-        native.state.init<Released>();
+        if (native.state.is<Accepted>()) {
+          // Native WebSocket no longer needed; release.
+          native.state.init<Released>();
+        } else if (native.state.is<Released>()) {
+          // While we were awaiting the jsg::Promise, someone else released our state. That's fine.
+        } else {
+          KJ_FAIL_ASSERT("Unexpected native web socket state", native.state.which());
```

Comment:
Looks like `kj::OneOf<...>::which()` returns a type that lacks the ability to be easily stringified, which causes a build error... Might be possible to do a static_cast to int first, if the value is useful to have in the message?  Or could use a switch and explicitly enumerate the other states.

Reply:
Ah crap, I verified that it went to int but failed to remember that `enum class` does not auto convert. I guess the switch is probably my favorite of those options!

Reply:
Looks OK to me.  I wasn't sure if the `KJ_CASE_ONEOF()` macro was OK with using statements instead of blocks, but reading its implementation, it seems like it should work.  Might be more consistent with other code to use blocks, though, as well as maybe playing nicer with any automated formatting tools.

Looks like the mac builder failure is due to some network error during setup?  Not sure why the windows and ubuntu builds were cancelled, though.

Reply:
Yeah, I'm actually a fan of braces with `break` even with return statements but I've had enough people ask for single line return here that I default to that.